parlai/tasks/triviaqa/agents.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+from parlai.core.dialog_teacher import DialogTeacher
+from parlai.core.agents import MultiTaskTeacher
+from .build import build
+
+import copy
+import json
+import os
+import random
+
+def _path(opt):
+    build(opt)
+
+    return (os.path.join(opt['datapath'], 'TriviaQA', 'qa'),
+            os.path.join(opt['datapath'], 'TriviaQA', 'evidence'))
+
+def setup_data_common(dataset, path, suffix, evidence_dir):
+    dataset_path = os.path.join(path,
+                                dataset + '-' + suffix + '.json')
+    print('loading: ' + dataset_path)
+    with open(dataset_path) as data_file:
+        data = json.load(data_file)['Data']
+    for datapoint in data:
+        question = datapoint['Question']
+        answers = datapoint['Answer']['Aliases']
+        if dataset == 'web':
+            evidence_list = datapoint['SearchResults']
+        else:
+            evidence_list = datapoint['EntityPages']
+
+        if len(evidence_list) == 0:
+            continue
+
+        evidence_num = random.randrange(len(evidence_list))
```

Comment:
adjust this as we discussed on messenger, thanks! (separate ex for each web evidence, concat all wiki evidence)